lib/puppet/gettext/config.rb

```diff
@@ -214,10 +219,10 @@ def self.load_translations(project_name, locale_dir, file_format)
   # @param [String] project_name the name of the project for which to load translations
   # @param [String] locale_dir the path to the directory containing translations
   # @param [Symbol] file_format the fomat of the translations files, :po or :mo
-  def self.add_repository_to_domain(project_name, locale_dir, file_format)
+  def self.add_repository_to_domain(project_name, locale_dir, file_format, text_domain = FastGettext.text_domain)
     return if @gettext_disabled || !gettext_loaded?
 
-    current_chain = FastGettext.translation_repositories[FastGettext.text_domain].chain
+    current_chain = FastGettext.translation_repositories[text_domain].chain
```

Comment:
Does it make sense to add a nil check here, or is that unnecessarily defensive?

Reply:
What would we do if it were nil?

Reply:
When I wrote this I was on the fence but ultimately decided it was unnecessary.

Reply:
Yeah I think you're right. This method is sufficiently private that if it is passed nil for some reason, that's a real bug we should fix.